oauthproxy.go

```diff
@@ -887,6 +891,18 @@ func (p *OAuthProxy) AuthenticateOnly(rw http.ResponseWriter, req *http.Request)
 	rw.WriteHeader(http.StatusAccepted)
 }
 
+func (p *OAuthProxy) ExtractSessionOnly(rw http.ResponseWriter, req *http.Request) {
```

Comment:
Instead we could also extend the AuthenticateOnly method to handle errors in two ways. What would you prefer?